modules/csv/test/csv-loader.spec.js

```diff
@@ -194,3 +194,33 @@ test('CSVLoader#loadInBatches(no header, row format, prefix)', async t => {
 
   t.end();
 });
+
+test('CSVLoader#loadInBatches(sample.csv, no dynamicTyping)', async t => {
+  const iterator = await loadInBatches(CSV_SAMPLE_URL, CSVLoader, {
+    csv: {
+      TableBatch: ColumnarTableBatch,
+      dynamicTyping: false,
+      // We explicitly set the header, since without dynamicTyping the first
+      // row might be detected as a header (all values would be string)
+      header: false
+    }
+  });
+  t.ok(isIterator(iterator) || isAsyncIterable(iterator), 'loadInBatches returned iterator');
+
+  let batchCount = 0;
+  for await (const batch of iterator) {
+    t.comment(`BATCH ${batch.count}: ${batch.length} ${JSON.stringify(batch.data).slice(0, 200)}`);
+    t.equal(batch.length, 2, 'Got correct batch size');
+
+    t.ok(validateColumn(batch.data.column1, batch.length, 'string'), 'column 0 valid');
+    t.ok(validateColumn(batch.data.column2, batch.length, 'string'), 'column 1 valid');
+    t.ok(
+      validateColumn(batch.data.column3, batch.length, 'string'),
+      'column 2 is a string and is valid'
+    );
+
+    batchCount++;
+  }
+  t.equal(batchCount, 1, 'Correct number of batches received');
```

Comment:
I think you can skip the batch counting here, this test focuses on the option.

Reply:
The batch counting was important to detect the impact on header detection (detected 1 row and 1 header instead of 2 rows). I think that we should keep it, but no strong feelings in case you want it removed.

Reply:
Don't you want to count the number of rows instead of the number of batches in that case?

Reply:
True! will update.